manifests/client.pp

```diff
@@ -77,9 +77,12 @@
       ensure => $package_ensure,
       name   => $package_name,
     }
-    $req = Package['snmp-client']
-  } else {
-    $req = undef
+  }
+
+  $req = $::osfamily ? {
+    'RedHat'             => [Package[$snmp::params::package_name], Package['snmp-client']],
```

Comment:
`Package[$snmp::params::package_name]` will work so long as no one does `include ::snmp::client` without `include ::snmp` (which is completely valid).